### PR TITLE
[CI] first attempt - base level

### PR DESCRIPTION
**Pull request by led0nk**
I think we should try our first attempts for CI.
At first there's much room for improvement but i tried to add useful things early on.

#### Makefile
- renamed go actions from `fmt` to `gofmt`
- added `build` 
- added `run`
- added `localtest`, because `gotest` is used for workflow
- added `gomoddownload` for workflow
- added `install-gotools` + rule for workflow
- added `golint` for workflow
- added gotidy
- added some setup-paths

#### Workflows
- added trigger on `pull_request`
- added `go-versions`, which checks for go version based on go.mod file and latest available. We use this for later actions like testing. This should help being prepared to upgrade go-version (we don't have to change workflow-steps if we upgrade to latest version).
- added `setup-env`, which can be useful later on, if we want to integrate e.g. some tests with coverage
- added `lint` (golangci-lint) 
**NOTE:** I have to work on linting errors later on in this PR
- switched from typical tests to "matrix-tests", we only use the checked go.mod-version and latest version, which is provided by `go-versions`. Should be useful for later upgrades of go-version.
- added `checks` so we can escape early if e.g. formatting doesn't fit
- changed build

#### gitignore
- added `.gitignore` file to prevent pushing binaries

I guess there is plenty of room for improvement, but we should start somewhere.
Just tell me if there are any additional things we could use e.g. like crosscompiling etc.

### CODE REVIEW

**Comment by led0nk** [comment]
i added some linting fixes, please take a look because previously there were some conflicts

File: Makefile [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
+
+localtest: gofmt govet check-fmt
+	CGO_ENABLED=0 go test -v ./... -failfast
+
+gomoddownload:
+	go mod download -x
+
+install-gotools: $(TOOLS_DIR)
+	#install golangci-lint v1.57.2 (only for workflow)
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(TOOLS_DIR) v1.57.2
```
Comment (frzifus) [inline Makefile]: Can we bundle the versions somewhere? e.g. as variables on top of this Makefile or in a `Makefile.common` and import it?

File: Makefile [hunk 2]
```diff
@@ -1,14 +1,49 @@
...
+
+install-gotools: $(TOOLS_DIR)
+	#install golangci-lint v1.57.2 (only for workflow)
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(TOOLS_DIR) v1.57.2 
+
+golint:
+	$(LINT) run --verbose --allow-parallel-runners --timeout=10m 
+
+gotidy:
+	go mod tidy -compat=1.21
```
Comment (frzifus) [inline Makefile]: Same with the go version

**Review by frzifus — COMMENTED** [review]
Thanks for cleaning this up :)

File: Makefile (hunk 1 above)
Comment (led0nk) [reply Makefile]: I added `Makefile.Common` for setting up variables like versions/directories.
Should be a cleaner attempt for filling in features later on.

**Review by frzifus — APPROVED** [review]

**Pull request merged**